File: Layout/app/src/main/java/com/example/layout/MainActivity.java
package com.example.layout;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btnLinear = findViewById(R.id.btnLinear);
        Button btnTable  = findViewById(R.id.btnTable);
        Button btnGrid   = findViewById(R.id.btnGrid);
        Button btnFrame  = findViewById(R.id.btnFrame);

        btnLinear.setOnClickListener(v ->
                setContentView(R.layout.activity_linear));

        btnTable.setOnClickListener(v ->
                setContentView(R.layout.activity_table));

        btnGrid.setOnClickListener(v ->
                setContentView(R.layout.activity_grid));

        btnFrame.setOnClickListener(v ->
                setContentView(R.layout.activity_frame));
    }
}

File: Fragment/app/src/main/java/com/example/fragment/MainActivity.kt
package com.example.fragment

import android.os.Bundle
import android.widget.FrameLayout
import androidx.fragment.app.FragmentActivity
import androidx.fragment.app.commit

class MainActivity : FragmentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        
        // Create a container for the fragment
        val container = FrameLayout(this).apply {
            id = android.view.View.generateViewId()
        }
        setContentView(container)

        if (savedInstanceState == null) {
            supportFragmentManager.commit {
                setReorderingAllowed(true)
                add(container.id, MainFragment())
            }
        }
    }
}